renderdoc/driver/vulkan/wrappers/vk_cmd_funcs.cpp

```diff
@@ -7570,6 +7571,379 @@ void WrappedVulkan::vkCmdEndRendering(VkCommandBuffer commandBuffer)
+
+    for(uint32_t i = 0; i < infoCount; ++i)
+    {
+      maxPrimitives.push_back({});
+      auto &rangeInfo = maxPrimitives.back();
+
+      rangeInfo.resize(pInfos[i].geometryCount);
+      memcpy(rangeInfo.data(), ppMaxPrimitiveCounts[i], rangeInfo.size() * sizeof(uint32_t));
+    }
+    SERIALISE_ELEMENT(maxPrimitives);
+  }
+
+  SERIALISE_CHECK_READ_ERRORS();
+
+  if(IsReplayingAndReading())
+  {
+    size_t tempmemSize = sizeof(VkAccelerationStructureBuildGeometryInfoKHR) * infoCount;
+    for(uint32_t i = 0; i < infoCount; ++i)
+      tempmemSize += GetNextPatchSize(&pInfos[i]);
+
+    byte *memory = GetTempMemory(tempmemSize);
+    auto *unwrappedInfos = reinterpret_cast<VkAccelerationStructureBuildGeometryInfoKHR *>(memory);
+    memory += sizeof(VkAccelerationStructureBuildGeometryInfoKHR) * infoCount;
+
+    for(uint32_t i = 0; i < infoCount; ++i)
+      unwrappedInfos[i] = *UnwrapStructAndChain(m_State, memory, &pInfos[i]);
+
+    ObjDisp(commandBuffer)
+        ->CmdBuildAccelerationStructuresIndirectKHR(Unwrap(commandBuffer), infoCount,
+                                                    unwrappedInfos, pIndirectDeviceAddresses,
+                                                    pIndirectStrides, ppMaxPrimitiveCounts);
+  }
+
+  return true;
+}
+
+void WrappedVulkan::vkCmdBuildAccelerationStructuresIndirectKHR(
+    VkCommandBuffer commandBuffer, uint32_t infoCount,
+    const VkAccelerationStructureBuildGeometryInfoKHR *pInfos,
+    const VkDeviceAddress *pIndirectDeviceAddresses, const uint32_t *pIndirectStrides,
+    const uint32_t *const *ppMaxPrimitiveCounts)
+{
+  {
+    size_t tempmemSize = sizeof(VkAccelerationStructureBuildGeometryInfoKHR) * infoCount;
+    for(uint32_t i = 0; i < infoCount; ++i)
+      tempmemSize += GetNextPatchSize(&pInfos[i]);
+
+    byte *memory = GetTempMemory(tempmemSize);
+    auto *unwrappedInfos = reinterpret_cast<VkAccelerationStructureBuildGeometryInfoKHR *>(memory);
+    memory += sizeof(VkAccelerationStructureBuildGeometryInfoKHR) * infoCount;
+
+    for(uint32_t i = 0; i < infoCount; ++i)
+      unwrappedInfos[i] = *UnwrapStructAndChain(m_State, memory, &pInfos[i]);
+
+    SERIALISE_TIME_CALL(ObjDisp(commandBuffer)
+                            ->CmdBuildAccelerationStructuresIndirectKHR(
+                                Unwrap(commandBuffer), infoCount, unwrappedInfos,
+                                pIndirectDeviceAddresses, pIndirectStrides, ppMaxPrimitiveCounts));
+  }
+
+  if(IsActiveCapturing(m_State))
+  {
+    {
+      CACHE_THREAD_SERIALISER();
+
+      SCOPED_SERIALISE_CHUNK(VulkanChunk::vkCmdBuildAccelerationStructuresIndirectKHR);
+      Serialise_vkCmdBuildAccelerationStructuresIndirectKHR(ser, commandBuffer, infoCount, pInfos,
+                                                            pIndirectDeviceAddresses,
+                                                            pIndirectStrides, ppMaxPrimitiveCounts);
+
+      m_FrameCaptureRecord->AddChunk(scope.Get());
```

Comment:
Serialised commands don't go into the frame capture record. This command could be serialised at any time - before the captured frame for submission within it, or during the captured frame but not serialised. If you put this directly into the frame capture record it will be orphaned.

Commands should go into the record for the command buffer - see the other functions being serialised in this file.

Reply:
Ah so the parent `if` statement should be `if(IsCaptureMode(m_State))` too?